File: inf_bez/1_laba/laba1_f_bez/laba1/enhanced/lab1.py
import string

alphabet = string.ascii_lowercase
shift = 3


def crypt(text):
    shift = 3
    rez = ""
    for c in text:
        if c != ' ':
            i = alphabet.index(c)
            i = (i + shift + len(alphabet)) % len(alphabet)
            rez = rez + alphabet[i]
            shift = ord(alphabet[i]) - ord('a')
        else:
            rez = rez + " "
    return rez


def decrypt(text):
    shift = 3
    rez = ""
    for c in text:
        if c != ' ':
            i = alphabet.index(c)
            i = (i - shift + len(alphabet)) % len(alphabet)
            rez = rez + alphabet[i]
            shift = ord(c) - ord('a')
        else:
            rez = rez + " "
    return rez;


in_str = "hello world"
out_str = crypt(in_str)
print(out_str)
print(decrypt(out_str))

File: inf_bez/1_laba/laba1_f_bez/laba1/enhanced/lab1.java
public class lab1 {
	public static final String alphabet = "abcdefghijklmnopqrstuvwxyz";
	public static int shift;
	
	public String crypt(String text) {
		shift =3;
		String rez = "";
		for (char c : text.toCharArray()) {
			if (c != ' ') {
				int i = alphabet.indexOf(c);
				i = (i + shift + alphabet.length()) % alphabet.length();
				rez = rez + alphabet.charAt(i);
				shift=alphabet.charAt(i)-'a';
			} else
				rez = rez + " ";
		}
		return rez;
	}
	

	public String decrypt(String text) {
		shift =3;
		String rez = "";
		for (char c : text.toCharArray()) {
			if (c != ' ') {
				int i = alphabet.indexOf(c);
				i = (i -shift + alphabet.length()) % alphabet.length();
				rez = rez + alphabet.charAt(i);
				shift=c-'a';
			} else
				rez = rez + " ";
		}
		return rez;
	}

	public static void main(String[] args) {
		lab1 l=new lab1();
		String c_text=l.crypt("hello world");
		System.out.println(c_text);
		System.out.println(l.decrypt(c_text));
	}

}
